xapian-letor/configure.ac

```diff
@@ -385,6 +385,12 @@ AH_BOTTOM(
 ])
 
 AC_DEFINE([XAPIAN_LIB_BUILD], [1], [Workaround for avoiding sub-header included error])
+AC_DEFINE([BUILD_BACKEND_GLASS], [1], [Workaround for avoiding setting the conditionals])
+AC_DEFINE([BUILD_BACKEND_INMEMORY], [1], [Workaround for avoiding setting the conditionals])
+AC_DEFINE([BUILD_BACKEND_REMOTE], [1], [Workaround for avoiding setting the conditionals])
+AM_CONDITIONAL([BUILD_BACKEND_GLASS], true)
+AM_CONDITIONAL([BUILD_BACKEND_INMEMORY], true)
+AM_CONDITIONAL([BUILD_BACKEND_REMOTE], true)
```

Comment:
The discussion was intended to give you some ideas to go and investigate; we are unlikely to be able to simply solve problems for you, nor should we.

You'll need to use `XAPIAN_HAS_*` to wrap code, _and_ ensure that `Makefile.mk` is suitably adjusted (which probably involves removing the conditions) so that everything is built. You've clearly considered this, so: what happened when you tried it?